lib/galaxy/managers/hdas.py

```diff
@@ -137,7 +137,7 @@ def copy_ldda(self, history, ldda, **kwargs):
     # .... deletion and purging
     def purge(self, hda, flush=True):
         if self.app.config.enable_celery_tasks:
-            celery_app.send_task('galaxy.celery.tasks.purge_hda', kwargs={'hda_id': hda.id})
+            purge_hda.delay(hda_id=hda.id)
```

Comment:
```
Traceback (most recent call last):
  File "/Users/mvandenb/src/galaxy/.venv/bin/celery", line 8, in <module>
    sys.exit(main())
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/__main__.py", line 15, in main
    sys.exit(_main())
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/bin/celery.py", line 213, in main
    return celery(auto_envvar_prefix="CELERY")
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/bin/base.py", line 132, in caller
    return f(ctx, *args, **kwargs)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/bin/worker.py", line 320, in worker
    worker = app.Worker(
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/worker/worker.py", line 94, in __init__
    self.app.loader.init_worker()
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/loaders/base.py", line 111, in init_worker
    self.import_default_modules()
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/loaders/base.py", line 106, in import_default_modules
    return [self.import_task_module(m) for m in self.default_modules]
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/loaders/base.py", line 106, in <listcomp>
    return [self.import_task_module(m) for m in self.default_modules]
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/loaders/base.py", line 86, in import_task_module
    return self.import_from_cwd(module)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/loaders/base.py", line 92, in import_from_cwd
    return import_from_cwd(
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/utils/imports.py", line 100, in import_from_cwd
    return imp(module, package=package)
  File "/Users/mvandenb/src/galaxy/.venv/lib/python3.8/site-packages/celery/loaders/base.py", line 89, in import_module
    return importlib.import_module(module, package=package)
  File "/Users/mvandenb/.pyenv/versions/3.8.2/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/mvandenb/src/galaxy/lib/galaxy/celery/tasks.py", line 7, in <module>
    from galaxy.managers.hdas import HDAManager
  File "/Users/mvandenb/src/galaxy/lib/galaxy/managers/hdas.py", line 16, in <module>
    from galaxy.celery.tasks import purge_hda
ImportError: cannot import name 'purge_hda' from partially initialized module 'galaxy.celery.tasks' (most likely due to a circular import) (/Users/mvandenb/src/galaxy/lib/galaxy/celery/tasks.py)
```
I'll rebase this one out

Reply:
I am still working on it, please just hold on.